src/learner.cc

```diff
@@ -725,13 +702,16 @@ class LearnerImpl : public Learner {
     // setup
     cfg_["num_feature"] = common::ToString(mparam_.num_feature);
     CHECK(obj_ == nullptr && gbm_ == nullptr);
-    obj_.reset(ObjFunction::Create(name_obj_));
+    obj_.reset(ObjFunction::Create(&tparam_, name_obj_));
     obj_->Configure(cfg_.begin(), cfg_.end());
     // reset the base score
     mparam_.base_score = obj_->ProbToMargin(mparam_.base_score);
-    gbm_.reset(GradientBooster::Create(name_gbm_, cache_, mparam_.base_score));
+    gbm_.reset(GradientBooster::Create(name_gbm_,
+                                       static_cast<LearnerTrainParam* const>(&tparam_),
```

Comment:
I don't think we need `static_cast` to cast non-const pointer to const, since the conversion should occur automatically.